test_half_bounded_generates_endpoint is flaky

```
=================================== FAILURES ===================================
_____________________ test_half_bounded_generates_endpoint _____________________
[gw2] linux2 -- Python 2.7.14 /usr/bin/python2
Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-python-3.44.12-py2/tests/cover/test_float_nastiness.py", line 88, in test_half_bounded_generates_endpoint
    settings=settings(max_examples=10000))
  File "/build/python-hypothesis/src/hypothesis-python-3.44.12-py2/src/hypothesis/core.py", line 1128, in find
    raise NoSuchExample(get_pretty_function_description(condition))
NoSuchExample: No examples found of condition lambda x: x == -1.0
============== 1 failed, 2351 passed, 6 skipped in 98.33 seconds ===============
```

Version info:
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.14, pytest-3.3.1, py-1.5.0, pluggy-0.6.0 -- /usr/bin/python2
cachedir: .cache
rootdir: /build/python-hypothesis/src/hypothesis-python-3.44.12-py2, inifile: tox.ini
plugins: hypothesis-3.44.12, xdist-1.21.0, forked-0.2, flaky-3.4.0
```

--- a/tests/cover/test_arbitrary_data.py
+++ b/tests/cover/test_arbitrary_data.py
@@ -34,7 +34,7 @@ def test_conditional_draw(x, data):
 def test_prints_on_failure():
     @given(st.data())
     def test(data):
-        x = data.draw(st.lists(st.integers(), min_size=1))
+        x = data.draw(st.lists(st.integers(0, 10), min_size=2))
         y = data.draw(st.sampled_from(x))
         x.remove(y)
         if y in x:

--- a/tests/cover/test_float_nastiness.py
+++ b/tests/cover/test_float_nastiness.py
@@ -24,7 +24,7 @@ import pytest
 
 import hypothesis.strategies as st
 from hypothesis import find, given, settings
-from tests.common.debug import minimal
+from tests.common.debug import minimal, find_any
 from hypothesis.internal.compat import WINDOWS
 from hypothesis.internal.floats import float_to_int, int_to_float
 
@@ -84,15 +84,13 @@ def test_can_generate_interval_endpoints(l, r):
 
 
 def test_half_bounded_generates_endpoint():
-    find(st.floats(min_value=-1.0), lambda x: x == -1.0,
-         settings=settings(max_examples=10000))
-    find(st.floats(max_value=-1.0), lambda x: x == -1.0,
-         settings=settings(max_examples=10000))
+    find_any(st.floats(min_value=-1.0), lambda x: x == -1.0)
+    find_any(st.floats(max_value=-1.0), lambda x: x == -1.0)
 
 
 def test_half_bounded_generates_zero():
-    find(st.floats(min_value=-1.0), lambda x: x == 0.0)
-    find(st.floats(max_value=1.0), lambda x: x == 0.0)
+    find_any(st.floats(min_value=-1.0), lambda x: x == 0.0)
+    find_any(st.floats(max_value=1.0), lambda x: x == 0.0)
 
 
 @pytest.mark.xfail(

--- a/tests/cover/test_float_nastiness.py
+++ b/tests/cover/test_float_nastiness.py
@@ -21,6 +21,7 @@ import sys
 import math
 
 import pytest
+from flaky import flaky
 
 import hypothesis.strategies as st
 from hypothesis import find, given, settings
@@ -83,6 +84,7 @@ def test_can_generate_interval_endpoints(l, r):
     find(interval, lambda x: x == r, settings=settings(max_examples=10000))
 
 
+@flaky(max_runs=4, min_passes=1)
 def test_half_bounded_generates_endpoint():
     find_any(st.floats(min_value=-1.0), lambda x: x == -1.0)
     find_any(st.floats(max_value=-1.0), lambda x: x == -1.0)

